osu.Game/Screens/OnlinePlay/Playlists/PlaylistsReadyButton.cs

```diff
@@ -80,9 +86,18 @@ public override LocalisableString TooltipText
             }
         }
 
-        private bool enoughTimeLeft =>
-            // This should probably consider the length of the currently selected item, rather than a constant 30 seconds.
-            room.EndDate != null && DateTimeOffset.UtcNow.AddSeconds(30).AddMilliseconds(gameBeatmap.Value.Track.Length) < room.EndDate;
+        private bool enoughTimeLeft()
+        {
+            // TODO: This doesn't consider mods which apply variable rates, yet.
+            double rate = ModUtils.CalculateRateWithMods(mods.Value);
+
+            // We want to avoid users not being able to submit scores if they chose to not skip,
+            // so track length is chosen over playable length.
+            double trackLength = Math.Round(gameBeatmap.Value.Track.Length / rate);
+
+            // Additional 30 second delay added to account for load and/or submit time.
+            return room.EndDate != null && DateTimeOffset.UtcNow.AddSeconds(30).AddMilliseconds(trackLength) < room.EndDate;
```

Comment:
This might also be tangential to the PR at best, but if a room does not have an end date then there is always enough time, right? I'm not sure if that ever happens in practice, but at least it sounds more correct on a conceptual level.

As is (and as it was previously), a room without end date always had no time left.